### PR TITLE
Keycloak

### PR DESCRIPTION
**Pull request by sdelamo**
To generate the guide locally: `./gradlew micronautOauth2KeycloakBuild`

Sample applications are generated at build/code

HTML is generated at:

`open build/dist/micronaut-oauth2-keycloak.html`

### CODE REVIEW

**Comment by sdelamo** [comment]
@morki Please provide PRs against  `keycloak` branch as base.

**Comment by sdelamo** [comment]
Implement this guide use https://www.keycloak.org 18.0 

See: https://github.com/micronaut-projects/micronaut-security/issues/1024

**Comment by morki** [comment]
Oh, sorry @sdelamo, I totally forget about this one.

**Comment by sdelamo** [comment]
@morki this is ready in case you want to review.

File: guides/micronaut-oauth2-keycloak/micronaut-oauth2-keycloak.adoc [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+dependency:micronaut-security-jwt[groupId=io.micronaut.security]
+
+
+=== OAuth 2.0 Configuration
+
+Add the following OAuth2 Configuration:
+
+resource:application-dev.yml[tag=oauth2]
+
+callout:authentication-idtoken[number=1,arg0=Keycloack]
```
Comment (wetted) [inline guides/micronaut-oauth2-keycloak/micronaut-oauth2-keycloak.adoc]:
```suggestion
callout:authentication-idtoken[number=1,arg0=Keycloak]
```

**Review by wetted — CHANGES_REQUESTED** [review]

**Review by wetted — APPROVED** [review]

**Pull request merged**